ngraph/core/reference/include/ngraph/runtime/reference/fake_quantize.hpp

```diff
@@ -0,0 +1,227 @@
+                bool out_low_trivial_broadcast = false;
+                bool out_high_trivial_broadcast = false;
+                bool in_low_aligned = false;
+                bool in_high_aligned = false;
+                bool out_low_aligned = false;
+                bool out_high_aligned = false;
+
+                auto check_trivial_broadcast =
+                    [&arg_shape, &arg_memory_offsets](Shape& shape_to_check,
+                                                      std::vector<size_t>& target_offsets,
+                                                      bool& trivial_broadcast,
+                                                      bool& aligned) {
+                        if (shape_size(shape_to_check) == 1 || shape_size(shape_to_check) == 0)
+                        {
+                            trivial_broadcast = true;
+                        }
+                        else if (shape_to_check == arg_shape)
+                        {
+                            aligned = true;
+                        }
+                        else
+                        {
+                            target_offsets =
+                                calc_broadcast_index_offset(arg_memory_offsets, shape_to_check);
+                        }
+                    };
+                check_trivial_broadcast(
+                    in_low_shape, in_low_offsets, in_low_trivial_broadcast, in_low_aligned);
+                check_trivial_broadcast(
+                    in_high_shape, in_high_offsets, in_high_trivial_broadcast, in_high_aligned);
+                check_trivial_broadcast(
+                    out_low_shape, out_low_offsets, out_low_trivial_broadcast, out_low_aligned);
+                check_trivial_broadcast(
+                    out_high_shape, out_high_offsets, out_high_trivial_broadcast, out_high_aligned);
+
+                std::vector<size_t> current_dim(arg_shape.size(), 0);
+
+                auto get_value = [&current_dim](bool is_trivial_broadcast,
+                                                bool is_aligned,
+                                                const T* data,
+                                                size_t idx,
+                                                const std::vector<size_t>& offsets) {
+                    T val;
+                    if (is_aligned)
+                    {
+                        val = data[idx];
+                    }
+                    else if (is_trivial_broadcast)
+                    {
+                        val = data[0];
+                    }
+                    else
+                    {
+                        size_t index_offset = calc_full_broadcast_offset(current_dim, offsets);
+                        idx -= index_offset;
+                        NGRAPH_CHECK(idx >= 0 && index_offset < shape_size(offsets),
+                                     "Incorrect index offset value!");
+                        val = data[idx - index_offset];
```

Comment:
can you confirm that?

Reply:
I'm not pretty sure yet but looks like it shouldn't work in such a case.
I'm going to revise broadcast logic since there is one more known issue with broadcast. Recently a lot of new low precision transformation tests were merged, and they rely on FQ references, so we can't merge FQ reference as is with disabling a few tests as it was planned initially.

Reply:
This code looks really strange

Reply:
This part is rewritten in the scope of broadcast issue fixing in this reference implementation